design/one-pager-terrajet-metadata-extraction.md

```diff
@@ -0,0 +1,272 @@
+            sku: '- (Required) SKU for the Analysis Services Server. Possible values are: D1, B1, B2, S0, S1, S2, S4, S8, S9, S8v2 and S9v2.'
+            timeouts:
+                create: '- (Defaults to 30 minutes) Used when creating the Analysis Services Server.'
+                delete: '- (Defaults to 30 minutes) Used when deleting the Analysis Services Server.'
+                read: '- (Defaults to 5 minutes) Used when retrieving the Analysis Services Server.'
+                update: '- (Defaults to 30 minutes) Used when updating the Analysis Services Server.'
+        # import statement scraped from the Terraform registry, if available
+        # Can be used for advanced purposes, such as constructing resource config "ExternalName.GetIDFn" functions, etc.
+        importStatements:
+            - terraform import azurerm_analysis_services_server.server /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourcegroup1/providers/Microsoft.AnalysisServices/servers/server1
+```
+
+Another alternative could be to have qualified names under the `argumentDocs`
+with a flat hierarchy, e.g., instead of a nested `ipv4_firewall_rule` block
+represented as a map, we could have its block parameters qualified with the
+configuration block name (`ipv4_firewall_rule.range_start`,
+`ipv4_firewall_rule.range_end`, etc.) Then `argumentDocs` would become a simple
+`map[string]string`.
+
+Another alternative could be to have per-resource YAML metadata files, i.e.,
+instead of the `resources` map in a single file, we could have each of its keys
+(and associated metadata) stored in a resource specific YAML-formatted file.
+These resource specific files could each be named as `<Terraform resource
+type>.yaml`, e.g., `azurerm_analysis_services_server.yaml`. 
+
+### Metadata scrapers
+Although not validated on all of available Terraform providers, at least, the
+big three Terraform providers ([terraform-provider-aws],
+[terraform-provider-azurerm] and [terraform-provider-google]) all have Terraform
+registry content in their respective repositories and use markdown documents
+with a common structure. Our assumption is that Terraform registry website is
+also generated using these markdown files:
+- For `terraform-provider-aws`:
+  https://github.com/hashicorp/terraform-provider-aws/tree/main/website/docs/r
+- For `terraform-provider-azurerm`:
+  https://github.com/hashicorp/terraform-provider-azurerm/tree/main/website/docs/r
+- For `terraform-provider-google`:
+  https://github.com/hashicorp/terraform-provider-google/tree/main/website/docs/r
+
+Thus a common metadata scraper implementation can extract metadata from these
+well-formatted per-resource markdown documents. Any spotted errors can then
+potentially be corrected manually in the scraped YAML metadata document.
```

Comment:
How would we prevent the scraper from overriding the manual changes made to the YAML metadata document?

Reply:
Added a discussion to the relevant section but also discussed this in the `Future Considerations` section.